### PR TITLE
Dev/rostik b/mks 90 additional task

### PR DESCRIPTION
**Pull request by Rostik18**

### CODE REVIEW

File: Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs [hunk 1]
```diff
@@ -39,21 +40,31 @@ public async Task<IActionResult> GetBand(int id)
         [HttpPut("{id}")]
         public async Task<IActionResult> PutBand(int id, Band band)
         {
-            return Ok();
+            var _band = await _bandService.PutBand(id, band);
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: нижній деш використовуєтсья тільки в неймінгу приватних змінних і ще деколи, але не тут

Comment (bvzayats) [reply Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: Put method should have DTO as a parameter, not Entity

File: Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs [hunk 2]
```diff
@@ -39,21 +40,31 @@ public async Task<IActionResult> GetBand(int id)
...
+
+            if (_band == null)
+                return BadRequest();
+
+            return Ok(band);
         }
 
         // POST: api/Bands
         [HttpPost]
         public async Task<IActionResult> PostBand(Band band)
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: Same

File: Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs (hunk 1 above)
Comment (Rostik18) [reply Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: Як буде краще назвати: localBand, updatedBand?

File: Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs [hunk 3]
```diff
@@ -39,21 +40,31 @@ public async Task<IActionResult> GetBand(int id)
...
         {
-            return Ok();
+            return Ok(await _bandService.PostBand(band));
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: Split to variable

File: Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs [hunk 4]
```diff
@@ -39,21 +40,31 @@ public async Task<IActionResult> GetBand(int id)
...
         }
 
         // DELETE: api/Bands/5
         [HttpDelete("{id}")]
         public async Task<IActionResult> DeleteBand(int id)
         {
-            return Ok();
+            var band = await _bandService.DeleteBand(id);
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: delete service method shouldn't return any data

File: Rostik_Baitsar/Festival/Festival/Servises/BandService.cs [hunk 5]
```diff
@@ -25,5 +25,45 @@ public async Task<Band> GetBand(int id)
...
+        }
+
+        public async Task<Band> DeleteBand(int id)
+        {
+            var band = await _context.Bands.FindAsync(id);
+
+            _context.Bands.Remove(band);
+            await _context.SaveChangesAsync();
+
+            return band;
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Servises/BandService.cs]: For what you return it?

Comment (Rostik18) [reply Rostik_Baitsar/Festival/Festival/Servises/BandService.cs]: How do I check if the removal is successful?
i need the service to return something?

File: Rostik_Baitsar/Festival/Festival/Models/BandDTO.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Festival.Models
+{
+    public class BandDTO
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public DateTime PerformanceTime { get; set; }
+
+        public static Band ConvertToEntity(BandDTO bandDTO)
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Models/BandDTO.cs]: It is awful solution

File: Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs [hunk 7]
```diff
@@ -37,22 +38,31 @@ public async Task<IActionResult> GetBand(int id)
 
         // PUT: api/Bands/5
         [HttpPut("{id}")]
-        public async Task<IActionResult> PutBand(int id, Band band)
+        public async Task<IActionResult> PutBand(int id, BandDTO bandDTO)
         {
-            return Ok();
+            var localBand = await _bandService.PutBand(id, BandDTO.ConvertToEntity(bandDTO));
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: use automapper, or initialize new instanse of Entity data type

File: Rostik_Baitsar/Festival/Festival/Models/Band.cs [hunk 8]
```diff
@@ -7,6 +7,5 @@ public class Band
         public int Id { get; set; }
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Models/Band.cs:7]: In the task is requirement min 5 fields in the table

File: Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs [hunk 9]
```diff
@@ -37,22 +38,31 @@ public async Task<IActionResult> GetBand(int id)
...
+            return Ok(localBand);
         }
 
         // DELETE: api/Bands/5
         [HttpDelete("{id}")]
         public async Task<IActionResult> DeleteBand(int id)
         {
+            _bandService.DeleteBand(id);
+
             return Ok();
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: Delete action should return NoContent

File: Rostik_Baitsar/Festival/Festival/Models/BandDTO.cs (hunk 6 above)
Comment (Rostik18) [reply Rostik_Baitsar/Festival/Festival/Models/BandDTO.cs]: How can I do better?

File: Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs (hunk 9 above)
Comment (Rostik18) [reply Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: changed

File: Rostik_Baitsar/Festival/Festival/Models/Band.cs (hunk 8 above)
Comment (Rostik18) [reply Rostik_Baitsar/Festival/Festival/Models/Band.cs:7]: Add 2 new fields

File: Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs (hunk 7 above)
Comment (Rostik18) [reply Rostik_Baitsar/Festival/Festival/Controllers/BandsController.cs]: used automapper

File: Rostik_Baitsar/Festival/Festival/Servises/IBandService.cs [hunk 10]
```diff
@@ -9,5 +9,11 @@ public interface IBandService
         Task<List<Band>> GetBands();
 
         Task<Band> GetBand(int id);
+
+        Task<Band> PutBand(int id, Band band);
+
+        Task<Band> PostBand(Band band);
+
+        void DeleteBand(int id);
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Servises/IBandService.cs]: Make Delete async

File: Rostik_Baitsar/Festival/Festival/Servises/BandService.cs [hunk 11]
```diff
@@ -25,5 +25,43 @@ public async Task<Band> GetBand(int id)
...
+
+        public async Task<Band> PostBand(Band band)
+        {
+            _context.Bands.Add(band);
+            await _context.SaveChangesAsync();
+
+            return band;
+        }
+
+        public void DeleteBand(int id)
```
Comment (bvzayats) [inline Rostik_Baitsar/Festival/Festival/Servises/BandService.cs]: Make it async

Comment (Rostik18) [reply Rostik_Baitsar/Festival/Festival/Servises/BandService.cs]: Done

File: Rostik_Baitsar/Festival/Festival/Servises/IBandService.cs (hunk 10 above)
Comment (Rostik18) [reply Rostik_Baitsar/Festival/Festival/Servises/IBandService.cs]: Done

**Review by bvzayats — APPROVED** [review]

**Review by Magfiozy — APPROVED** [review]

**Pull request merged**